### PR TITLE
[Common] Move async class to common and support in executor.

### PR DESCRIPTION
**Pull request by q82419**
This is for WasmEdge `0.13.0`.

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [serverless review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/). Here are my reviews of code commits in this PR.

------

File: include/common/async.h [hunk 1]
```diff
@@ -66,20 +65,15 @@ template <typename T> class Async {
...
-    swap(LHS.VMPtr, RHS.VMPtr);
+    swap(LHS.StopFuncPtr, RHS.StopFuncPtr);
   }
 
-  void cancel() noexcept {
-    if (likely(VMPtr)) {
-      VMPtr->stop();
-    }
-  }
+  void cancel() noexcept { StopFuncPtr(); }
```
Comment (ibmibmibm) [inline include/common/async.h:71]:
```c++
void cancel() noexcept {
  if (likely(StopFuncPtr)) {
    StopFuncPtr();
  }
}
```

**Review by ibmibmibm — CHANGES_REQUESTED** [review]

File: include/common/async.h (hunk 1 above)
Comment (q82419) [reply include/common/async.h:71]: The `std::function` cannot support this.

**Comment by juntao** [comment]
flows review

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: include/common/async.h (hunk 1 above)
Comment (q82419) [reply include/common/async.h]: Use `operator bool()` instead.

**Comment by q82419** [comment]
@apepkuss Please check that the API is what you want. Thanks.

**Comment by apepkuss** [comment]
@q82419 Thanks for the message. I'll check the new APIs. Thanks!

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by ibmibmibm — APPROVED** [review]

**Pull request merged**